### PR TITLE
Nitiwiki: fix isues with relative paths and the root href

### LINKED ISSUE #1872: nitiwiki: relative links for auto index
Auto-index relative links (`SectionIndex::href`) are broken.

Generated links from the index contains the section href twice.

@privat The problem seems to bere there since your PR about rellinks (#1505) have fun! :)

### PR DESCRIPTION
**Pull request by privat**
close #1872 

Can you try it @Morriar ?

### CODE REVIEW

File: contrib/nitiwiki/src/wiki_links.nit [hunk 1]
```diff
@@ -100,12 +100,23 @@ redef class WikiEntry
 	# Relative path to `self` from the target root_url
 	fun href: String do return breadcrumbs.join("/")
 
+	# Relative path to the directory `self` from the target root_url
+	fun dhref: String do return href.dirname
```
Comment (Morriar) [inline contrib/nitiwiki/src/wiki_links.nit]: I would prefer something like `dir`.

**Comment by Morriar** [comment]
The `dref` stills bother me but +1

**Comment by xymus** [comment]
+1

**Comment by privat** [comment]
reroll: renamed `dhref` to `dir_href`

**Pull request merged**